std/range/package.d

```diff
@@ -4125,32 +4125,60 @@ pure unittest
     Generate lockstep's opApply function as a mixin string.
     If withIndex is true prepend a size_t index to the delegate.
 */
-private string lockstepMixin(Ranges...)(bool withIndex)
+private string lockstepMixin(Ranges...)(bool withIndex, bool reverse)
 {
     import std.format : format;
 
     string[] params;
     string[] emptyChecks;
     string[] dgArgs;
     string[] popFronts;
+    string indexDef;
+    string indexInc;
 
     if (withIndex)
     {
         params ~= "size_t";
         dgArgs ~= "index";
+        if (reverse)
+        {
+            indexDef = q{
+                size_t index = ranges[0].length-1;
+                enforce(_stoppingPolicy == StoppingPolicy.requireSameLength, "lockstep can only be used with foreach_reverse when stoppingPolicy == requireSameLength");
```

Comment:
Hmm convinced
1) we don't want to have different behavior
2) we don't want unexpected runtime overhead

> This is the current behavior:

`lockstep` uses `shortest` by default, thus currently with `foreach` it won't touch the `3`.

```
foreach (e1, e2; lockstep([1, 2, 3], [4, 5])) {
 assert(e1 == e2 - 3);
}
```

So when reverse-iterating I would expect a similar behavior - and starting with `3` (and ignoring the `1`) as it currently would do seems pretty consistent to me.

Regardless we probably should ask for a different opinion. Anyone?